python/pyiceberg/io/pyarrow.py

```diff
@@ -470,13 +471,54 @@ def expression_to_pyarrow(expr: BooleanExpression) -> pc.Expression:
     return boolean_expression_visit(expr, _ConvertToArrowExpression())
 
 
+def _file_to_table(
```

Comment:
The reason why I passed `row_filter` and `table` to this method in my branch was so that we could make the method public. I think we are going to need that so that tasks can be distributed to other nodes or processes and read independently. Having a method that reads a task and another that reads an iterable of tasks in a threadpool seems like a good API to provide.

Reply:
Shall we do that in a separate PR? I also gave multi-processing a shot, but I ran into a lot of serialization issues. Having this private until we fix all of that prevents us from changing public APIs later on.